### DISCUSSION TITLE
How to get internal data for a custom block

### DISCUSSION THREAD

**Discussion by xiaoxiji1234** [lowdefy/lowdefy#967] [Questions and Help] [answered]
Hello, I want to get the internal data of custom block in lowdefy. Is there any method?

**Comment by Gervwyk** [comment]
Hi @xiaoxiji1234 

Could you perhaps give a code example of your custom block for some more context?

**Comment by xiaoxiji1234** [comment, marked as answer]
Hi，@Gervwyk.  I want to get the return value of this method in lowdefy. Methods.registerMethod can call this function in lowdefy, but it can not get the return value. What is the method?
block.jsx
```js
// 保存页面
  const savePage = async () => {
    const yamlStr = parseToYaml();
    const params = {
      appName: appName.current,
      content: yamlStr,
      pageName: pageData[pageIndex.current].id,
    };
    return params
  };
methods.registerMethod('save',savePage)
```

**Reply by Gervwyk** [reply]
So the response of the evaluated method is accessible for the preceding actions in the event. You can access this response value in the following actions using the [`_actions`](https://docs.lowdefy.com/_actions) operator.

However, you are looking to use this response value in your block. The closest thing to this is what we do for the [`S3UploadButton`](https://docs.lowdefy.com/S3UploadButton) you can look at the code here: https://github.com/lowdefy/lowdefy/blob/main/packages/blocks/blocksAntd/src/blocks/S3UploadButton/S3UploadButton.js
(This implementation is a little hacky and will likely change in the future)

Alternatively consider using `methods.registerMethod` or `methods.setValue` instead of `methods.registerMethod`. Can you elaborate on why you decided to go with  `methods.registerMethod` for your use case?

Note, for `methods.setValue` your block category should be an `input`.

**Reply by xiaoxiji1234** [reply]
OK, it has been solved. Thank you very much.
